### DISCUSSION TITLE
MySQL Table Relationships

### DISCUSSION THREAD

**Discussion by timmehrees** [Budibase/budibase#10577] [Help] [answered]
Hi All,

In the last update or so my table relationships appear to have gone heywire, and i've had to attempt to recreate them. However when I try to recreate the connection I get the error "Column name cannot be an existing column" - Which makes sense given the data already exists.

I'm running MySQL as the database, and i've only got 2x One-to-many relationships. It appears in PHPMyAdmin that the original relationship (fk's etc) is still established, but not working for pulling data into forms etc. (queries where I have directly referenced the fk etc all work).

I don't think this is a bug, I think I'm just missing something, anyone have any ideas?

Thanks,
Tim

**Comment by melohagan** [comment]
Hey @timmehrees 

Would you be able to share some screenshots please?

**Reply by timmehrees** [reply]
This was initially set up as a one-to-one - as that is no longer an option, I suspect that's why it broke...

Data in the field - matches the correct id from the other table.
<img width="218" alt="Screen Shot 2023-05-20 at 15 45 54" src="https://github.com/Budibase/budibase/assets/7864923/b1f02c02-0a02-4935-95d4-966fdca9d23d">


The error when I attempt to connect it back up - Its not wrong... but that doesn't help me reconnect it.
<img width="344" alt="Screen Shot 2023-05-20 at 15 45 14" src="https://github.com/Budibase/budibase/assets/7864923/ad74da77-af2a-4407-bbbb-99ce605060bb">



Thanks again.

**Reply by melohagan** [reply, marked as answer]
Thanks @timmehrees 

So the only thing that changed was the label. The previous labels were `One` and `Many`. These were updated to `One to Many` and `Many to Many` to be more explicit about the type of relationship being defined, but the functionality is the same.

There is no specific One->One option. For that case you can use One->Many.

The *From table column* field specifies the column name of the relationship field within Budibase. This name must be unique. You are getting the error because you are trying to give it the same name as the foreign key.

For example:
![Screenshot 2023-06-02 at 15 15 27](https://github.com/Budibase/budibase/assets/101575380/5bf5a183-db9f-4d00-82b0-bd79f5441e2a)

![Screenshot 2023-06-02 at 15 17 00](https://github.com/Budibase/budibase/assets/101575380/d51763bd-1db6-4244-90bb-93f26b269856)

The `fk_owner` field exists on the Postgres table. The `Owners` field is a relationship field that only exists within Budibase.
